Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs

```diff
@@ -186,6 +186,7 @@ private IEnumerator UpdateState()
                 switch (state)
                 {
                     case KeyboardState.Showing:
+                        if (TouchScreenKeyboard.visible == true)
```

Comment:
I think this might bust some OS versions where the `.visible` state isn't properly reported. That's a main reason why we wrote all this custom keyboard code in the first place. @MaxWang-MS might be able to elaborate.

Reply:
Thanks for pointing this out Kurtis! It is a known Unity issue that the .visible state can be inaccurate under certain circumstances, even on the latest HL2 OS. So we deliberately decided to track the state ourselves in this class instead of relying on that Unity property.